src/tests/psyGen_test.py

```diff
@@ -810,3 +815,794 @@ def test_invalid_reprod_pad_size():
             "REPROD_PAD_SIZE in config.py should be a positive "
             "integer") in str(excinfo.value)
     config.REPROD_PAD_SIZE = keep
+
+
+def test_argument_dependent_arg():
+    '''Check that the dependent_arg method returns the appropriate boolean
+    value for arguments with combinations of read and write access'''
+    _, invoke_info = parse(os.path.join(BASE_PATH,
+                                        "4.5_multikernel_invokes.f90"),
+                           distributed_memory=False, api="dynamo0.3")
+    psy = PSyFactory("dynamo0.3", distributed_memory=False).create(invoke_info)
+    invoke = psy.invokes.invoke_list[0]
+    schedule = invoke.schedule
+    arg_f1_inc_1 = schedule.children[0].children[0].arguments.args[0]
+    arg_f1_inc_2 = schedule.children[2].children[0].arguments.args[0]
+    arg_f2_read_1 = schedule.children[0].children[0].arguments.args[2]
+    arg_f2_inc = schedule.children[1].children[0].arguments.args[0]
+    arg_f2_read_2 = schedule.children[2].children[0].arguments.args[1]
+    # different names returns False
+    assert not arg_f2_inc._dependent_arg(arg_f1_inc_1)
+    # same name both reads returns False
+    assert not arg_f2_read_1._dependent_arg(arg_f2_read_2)
```

Comment:
I agree about changing the name, so have done so.

The terminology is not standard as there does not appear to be terminology for what I need. What I did was borrow the terminology from loop dependence analysis where a forward dependence is one from a future loop iteration and a backward dependence is one from a previous loop iteration. I've hopefully explained this in enough detail in the doc string as requested.